src/click/utils.py

```diff
@@ -29,6 +30,27 @@
 R = t.TypeVar("R")
 
 
+class Sentinel(enum.Enum):
+    """Enum used to define sentinel values.
+
+    .. seealso::
+
+        `PEP 661 - Sentinel Values <https://peps.python.org/pep-0661/>`_.
+    """
+
+    UNSET = object()
+
+    def __repr__(self) -> str:
+        return f"{self.__class__.__name__}.{self.name}"
+
+
+UNSET = Sentinel.UNSET
```

Comment:
This should be falsy, right?

Reply:
That doesn't address my question, though. `Sentinel.UNSET` should be falsy then:

```python
>>> from click.utils import UNSET
>>> value = UNSET
>>> if not UNSET:
...     print('No value provided!')
... else:
...     print(value)
...
Sentinel.UNSET
```